src/common/Utilities.py

```diff
@@ -463,3 +463,29 @@ def str_to_int_list(str_list):
             int_item = list(range(strt, stop + 1))
         int_list = int_list + int_item
     return int_list
+
+def dot_product_test(operator, num_tests = 20, max_err = 10e-5):
+    '''
+    Test if a given operator is adjoint.
+    The operator needs to have been already set_up() with valid AcquisitionModel and ImageData objects.
+    The operator needs to have methods direct() and adjoint() implemented
+    '''
+    # Get image and data to know the sizes of each operation.
+    image=operator.domain_geometry()
+    image.fill(1.0)
+    data=operator.direct(image) # Is there a better way than this? Similar to .domain_geomerty() ?
```

Comment:
currently we don't have one in SIRF I guess. @paskino? You could do this inside the loop to avoid having to do one extra `direct`

Reply:
Ah pity, its much cleaner if it is out of the loop.

Reply:
sure. possibly we have `range_geometry` I sort of thought CIL would need it.

By the way, if the objects are allocated upfront, you might be able to save some mememory by using the `out` parameter